src/html/generator.ml

```diff
@@ -406,12 +406,10 @@ module Page = struct
     Utils.list_concat_map ~f:(include_ ~config) subpages
 
   and page ~config p : Odoc_document.Renderer.page list =
-    let { Page.preamble; items = i; url } = Doctree.Labels.disambiguate_page p
-    and subpages =
-      (* Don't use the output of [disambiguate_page] to avoid unecessarily
-         mangled labels. *)
-      subpages ~config @@ Doctree.Subpages.compute p
```

Comment:
We use the output of `disambiguate_pages`, but a flag is passed to this function to avoid unnecessary mangled labels.
We could add a comment along the lines of:
> Do not enter subpages when disambiguating to avoid unecessarily mangled labels

Reply:
You are right, I missed that. No need to add a comment now that the interesting part is gone.